Services/DataX.Utilities/DataX.Utilities.Web/RolesCheck.cs

```diff
@@ -25,6 +30,30 @@ public static void EnsureWriter(HttpRequest request, bool isLocal)
             }
         }
 
+        /// <summary>
+        /// A helper method that Adds the test client user id to the white list from keyvault if it exists
+        /// TODO: Support adding this whitelist on Kubernetes using IConfiguration object
+        /// </summary>
+        private static void AddWhitelistedTestClientUserId()
+        {
+            if (HostUtil.InServiceFabric)
+            {                
+                var serviceKeyvaultName = ServiceFabricUtil.GetServiceKeyVaultName().Result.ToString();
+                var configPackage = FabricRuntime.GetActivationContext().GetConfigurationPackageObject("Config");
+                var serviceEnvironmenConfig = configPackage.Settings.Sections["ServiceEnvironment"];                
+                var testClientId = serviceEnvironmenConfig.Parameters["TestClientId"].Value;
+                try
+                {
+                    _ClientWhitelist.Add(KeyVault.KeyVault.GetSecretFromKeyvault(serviceKeyvaultName, testClientId));

```

Comment:
there is a KeyVault.TryResolveSecret...()..

Reply:
we discussed that KeyVault.TryResolveSecret() also throws. For now, catching the execption, if any, and logging the error in AppInsights.